src/sdk/namespace_s3.js

```diff
@@ -148,21 +148,24 @@ class NamespaceS3 {
             const request = {
                 Bucket: this.bucket,
                 Key: params.key,
-                PartNumber: params.part_number,
+                PartNumber: params.multipart_number,
             };
-            // If set, part_number is positive integer from 1 to 10000.
+            // If set, multipart_number is positive integer from 1 to 10000.
             // Usually part number is not provided and then we read a small "inline" range
             // to reduce the double latency for small objects.
             // can_use_get_inline - we shouldn't use inline get when part number exist or when heading a directory
-            const can_use_get_inline = !params.part_number && !request.Key.endsWith('/');
+            const can_use_get_inline = !params.multipart_number && !request.Key.endsWith('/');
             if (can_use_get_inline) {
                 request.Range = `bytes=0-${config.INLINE_MAX_SIZE - 1}`;
             }
             this._set_md_conditions(params, request);
             this._assign_encryption_to_request(params, request);
             let res;
             try {
-                res = can_use_get_inline ?
+                // params.multipart_number was added, since head
+                // does not return Content-Range, which is needed
+                // for GetObject with part num.
+                res = (can_use_get_inline || params.multipart_number) ?
```

Comment:
Well, [HeadObject](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html) does not contain Content-Range, get [GetObject](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html) does. Does it still look fishy 🐟 ?

Reply:
Yes, this code is an optimization path, which uses GET object instead of HEAD, when there is a chance that this is a small object and we can get the entire data in one trip.
So why does having partNumber in the request causes us to use GET?
We actually avoid GET when partNumber is provided - [here](https://github.com/noobaa/noobaa-core/blob/e04e45f2ab6236ce060a1ea26b0c63bee396aa2e/src/sdk/namespace_s3.js#L157).
So I still don't follow the new logic.

Reply:
@romayalon - Can you take a look at this change too - it's a delicate optimization logic and I hope I'm not confusing.

Reply:
@guymguym this is for setting the `Content-Range` field in the response. You could call me if you have more questions...

Reply:
Why is it needed for HEAD? I don't see it in S3 API docs -
https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html

Reply:
@guymguym The read_object_md() function is called both by GET and HEAD flows.  WDYT?

Reply:
I still don't see the relation of that fact to the change you made

Reply:
Please call me, hope I could explain it.